### PR TITLE
bump: Update sonar dotnet

### PR DESCRIPTION
**Pull request by MiguelMarcelino**

### CODE REVIEW

**Comment by albertoadami** [comment]
@MiguelMarcelino is this `sonar-csharp-plugin.jar` file expected to be here? If we need it maybe will be nicer  to have it in a `bin` directory

**Comment by MiguelMarcelino** [comment]
> @MiguelMarcelino is this `sonar-csharp-plugin.jar` file expected to be here? If we need it maybe will be nicer to have it in a `bin` directory

It is not. However, this branch is a draft for a reason: We will not be able just to bump the tool. This will require us to rework our parsing mechanism.

File: src/DocsGenerator/Program.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ private static void Main()
 
             var descriptions = new List<Description>();
 
-            var doc = XDocument.Load(@".res/sonar-csharp-rules.xml");
+            var doc = XDocument.Load(@".res/sonar-csharp-rules.json");
```
Comment (MiguelMarcelino) [inline src/DocsGenerator/Program.cs:39]: This is not supposed to be.

**Comment by MiguelMarcelino** [comment]
Closing this. A simple bump won't cut it.

**Pull request closed without merging**